Scorer

Izayuh opened the pull request:
Simple prototype for our scoring function.

Commit:
Add menu bar item

beszel committed:
Rename main file to `main.gs`

Izayuh committed:
Test

beszel committed:
Added test to read data from sheet

beszel committed:
Add constants and append test row to file

beszel committed:
Use var instead of const

beszel committed:
Output names and top 3 workshop preferences into output file.

beszel committed:
Fix improperly renamed variables

beszel committed:
Remove unneeded logging messages

Izayuh committed:
Added simple scoring function

Chris-Pyles committed:
Modified scorer function

Added to scorer function code that finds and outputs to the log students who received none of their top 6 matches

beszel committed:
Merge branch 'master' into Scorer

beszel committed:
Style changes

beszel commented on main.gs:
```diff
@@ -31,3 +31,44 @@ function matchGirls() {
+        }if((outputData[i][2] == responseData[i][COLUMN_PREFERENCE_6])
+            || (outputData[i][3] == responseData[i][COLUMN_PREFERENCE_6])
+            || (outputData[i][4] == responseData[i][COLUMN_PREFERENCE_6])) {
+            score += 12;
```
Write constants to remove these magic numbers. Can you refactor this block to reduce repetition? What about a helper function?

beszel commented on main.gs:
```diff
@@ -31,3 +31,44 @@ function matchGirls() {
         outputSheet.appendRow([firstName, lastName, preference_1, preference_2, preference_3]);
     }
 }
+
```
Add doctag

Izayuh committed:
Removed magic numbers.

Izayuh committed:
Update main.gs

Added simple comments to the matchgirls and scorer functions.

Chris-Pyles committed:
Updated scorer function and added function checkMatches

-scorer function restored to match Scorer branch
-checkMatches function adds to the end of the output data a sorted list describing the student's matches, and an additional row if they got none of their top 6 preferences.

beszel committed:
Extract `scorePreference` function from scorer

beszel commented:
If a student doesn't get any of their preferences, they get a perfect score of 0. They should get a worse score than that.

beszel committed:
Give bad scores for unwanted workshops, refactor

Chris-Pyles committed:
Updated main.gs

Updated other functions and variables to match most up-to-date versions
Finished checkMatches function that writes to the output sheet the matches that the student received, and appends "NO MATCHES" if they didn't get any of their top 6 preferences
Tested on students who got matches in top 3, only some of their top 3 matches, and none of their top 3 matches. Got correct results for all cases.

IMPORTANT NOTE: the checkMatches function accounts for students who listed a workshop more than once in their preferences. This will need to be accounted for in ALL other functions.

Izayuh committed:
Cleaned up the scorer function again, this time using code from checkMatches

Izayuh committed:
Update main.gs

Izayuh committed:
ACTUAL commit

Izayuh committed:
Merge branch 'Scorer' into FindWorst

Chris-Pyles committed:
Merge pull request #16 from WhitmanCSCapstone/FindWorst

Added checkMatches function

Chris-Pyles committed:
fixed errors between master and scorer branches

Variables missing from the scorer branch and present in the master branch were added.
All code was tested and functions properly.

Chris-Pyles committed:
Update main.gs

Chris-Pyles commented:
this SHOULD be set for approval now, but it says there are conflicts. I feel like this has to do with the fact that master branch has main.js and scorer has main.gs. We might want to make them the same filetype for the future.

Chris-Pyles committed:
Rename main.gs to main.js

this should fix the error in the merging of scorer and master

Chris-Pyles committed:
removed unnecessary TODO from line 69

nice

Chris-Pyles committed:
Merge branch 'master' into Scorer
